src/aggregation.c

```diff
@@ -202,13 +202,9 @@ static int agg_instance_create_name(agg_instance_t *inst, /* {{{ */
     else if (strcmp("", tmp_plugin) != 0)
```

Comment:
I agree that, as written, the final `else` is unreachable, but I believe this wasn't the intention of this code. It seems to cover 4 cases:
1. both `tmp_plugin` and `tmp_plugin_instance` are empty;
2. only `tmp_plugin_instance` is empty, but `tmp_plugin` is not;
3. only `tmp_plugin` is empty, but `tmp_plugin_instance` is not;
4. neither `tmp_plugin` nor `tmp_plugin_instance` is empty.

So, what @octo probably intended in https://github.com/collectd/collectd/commit/c7d2dc3967ca49886f2869d9a5dc8ce40a3b126e was:
```c
    if ((strcmp("", tmp_plugin) == 0) && (strcmp("", tmp_plugin_instance) == 0))
      sstrncpy(inst->ident.plugin_instance, AGG_FUNC_PLACEHOLDER,
               sizeof(inst->ident.plugin_instance));
    else if (strcmp("", tmp_plugin_instance) == 0)  // tmp_plugin is non-empty.
      snprintf(inst->ident.plugin_instance, sizeof(inst->ident.plugin_instance),
               "%s-%s", tmp_plugin, AGG_FUNC_PLACEHOLDER);
    else if (strcmp("", tmp_plugin) == 0)  // tmp_plugin_instance is non-empty.
      snprintf(inst->ident.plugin_instance, sizeof(inst->ident.plugin_instance),
               "%s-%s", tmp_plugin_instance, AGG_FUNC_PLACEHOLDER);
    else
      snprintf(inst->ident.plugin_instance, sizeof(inst->ident.plugin_instance),
               "%s-%s-%s", tmp_plugin, tmp_plugin_instance,
               AGG_FUNC_PLACEHOLDER);
               "%s-%s", tmp_plugin_instance, AGG_FUNC_PLACEHOLDER);
```
Should we fix it to match the intent here?

_(Sadly, this will nullify the side effect in your PR description...)_